### PR TITLE
Box-beam and PBP scanning notebook overhauls

### PR DESCRIPTION
**Pull request by jadball**
- Added new `FrelonSegmenterGui` in `nbGui/segmenter_gui.py`
- Refactored `nbGui/3DXRD/0_3DXRD_segment_frelon.ipynb` accordingly
- Overhauled all box-beam 3DXRD notebooks to use new PYTHONPATH script from @jonwright 
- Added box-beam grid indexing notebook to `nbGui/3DXRD/2_3DXRD_index_grid.py`
- Added Friedel pair indexing notebook from @jonwright to  `nbGui/3DXRD/2_3DXRD_index_friedel.py`
- Much better peak selection plots in `nbGui/3DXRD/1_3DXRD_refine_parameters.py` and `nbGui/3DXRD/2_3DXRD_index.py`
- We now run `makemap` again at the end of `nbGui/3DXRD/2_3DXRD_index.py` to get proper peak assignments against all 3d peaks, which are saved into `ds.col3dfile`
- Renamed PBP Scanning notebooks for clarity
- Added option to run PBP indexing on cluster (`nbGui/S3DXRD/pbp_1_indexing.ipynb`) - looks for a minimum 90-core machine in `nice-long`, see `nbGui/S3DXRD/run_pbp_recon.py` and new function `prepare_pbp_bash()` in `nbGui/nb_utils.py`
- Much faster and better PBP visualization (`nbGui/S3DXRD/pbp_2_visualise.ipynb`) using @jonwright `pbpmap` class from stainless steel datasets

### CODE REVIEW

File: ImageD11/nbGui/S3DXRD/run_pbp_recon.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+if __name__ == "__main__":
+    # horrible workaround to include id11 code path
+    import sys
+    
+    id11_code_path = sys.argv[1]
```
Comment (jonwright) [inline ImageD11/nbGui/S3DXRD/run_pbp_recon.py:5]: Better to the environment variable PYTHONPATH outside of the script. e.g.
`PYTHONPATH={id11_code_path} {sys.executable} run_pbp_recon.py ...`

File: ImageD11/nbGui/S3DXRD/run_pbp_recon.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    dstol = float(sys.argv[5])
+    etacut = float(sys.argv[6])
+    ifrac = float(sys.argv[7])
+    costol = float(sys.argv[8])
+    y0 = float(sys.argv[9])
+    symmetry = sys.argv[10]
+    foridx = [int(x) for x in ast.literal_eval(sys.argv[11])]
+    forgen = [int(x) for x in ast.literal_eval(sys.argv[12])]
+    uniqcut = float(sys.argv[13])
+    minpkint = int(sys.argv[14])
```
Comment (jonwright) [inline ImageD11/nbGui/S3DXRD/run_pbp_recon.py:26]: For lima segmenter we read the parameters from the dataset (there is a group), which is already the first arg. With several pbp runs (different phases, parameters etc), we could just create a group for each one that contains the parameters at attrs and the pbpfile and maps and members / links.

File: ImageD11/nbGui/S3DXRD/run_pbp_recon.py (hunk 1 above)
Comment (jadball) [reply ImageD11/nbGui/S3DXRD/run_pbp_recon.py]: Fixed in https://github.com/FABLE-3DXRD/ImageD11/pull/317/commits/6f3d38f9c70431ad1bc93b0fdb4dd44c93c48d1d

File: ImageD11/nbGui/S3DXRD/run_pbp_recon.py (hunk 2 above)
Comment (jadball) [reply ImageD11/nbGui/S3DXRD/run_pbp_recon.py]: I completely agree - if you don't mind, I think this should become a wider mini-project (saving indexing parameters to the dataset object) which we could make an issue for? and in the meantime merge this PR?

**Pull request merged**